osu.Game.Rulesets.Osu/Difficulty/Evaluators/SpeedEvaluator.cs

```diff
@@ -31,14 +30,22 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, double gr
             // derive strainTime for calculation
             var osuCurrObj = (OsuDifficultyHitObject)current;
             var osuPrevObj = current.Index > 0 ? (OsuDifficultyHitObject)current.Previous(0) : null;
+            var osuNextObj = (OsuDifficultyHitObject)current.Next(0);
 
             double strainTime = osuCurrObj.StrainTime;
             double greatWindowFull = greatWindow * 2;
-            double speedWindowRatio = strainTime / greatWindowFull;
+            double doubletapness = 1;
 
-            // Aim to nerf cheesy rhythms (Very fast consecutive doubles with large deltatimes between)
-            if (osuPrevObj != null && strainTime < greatWindowFull && osuPrevObj.StrainTime > strainTime)
-                strainTime = Interpolation.Lerp(osuPrevObj.StrainTime, strainTime, speedWindowRatio);
+            // Nerf doubletappable doubles.
+            if (osuNextObj != null)
+            {
+                double currDeltaTime = Math.Max(1, osuCurrObj.DeltaTime);
+                double nextDeltaTime = Math.Max(1, osuNextObj.DeltaTime);
+                double deltaDifference = Math.Abs(nextDeltaTime - currDeltaTime);
+                double speedRatio = Math.Min(1, currDeltaTime / deltaDifference);
```

Comment:
I would do `double speedRatio = currDeltaTime / Math.Max(currDeltaTime, deltaDifference);` to avoid div by 0.

Reply:
Isn't this a different calculation? Wouldn't the expression `max(currDeltaTime, deltaDifference)` tend towards the `currDeltaTime` value resulting in `currDeltaTime / currDeltaTime => 1`?

E.g. Suppose:
```
currDeltaTime = 10
nextDeltaTime = 15
deltaDifference = abs(15 - 10) = 5

---- Previous ----

speedRatio = min(1, 10 / 5) = 2

---- Proposal ----

speedRatio = 10 / max(10, 5) = 1
```

Reply:
I did Desmos this and run it through my head to check if the behaviour was identical and it seemed fine, but in retrospect... I'm blaming it on the heatwave today 😅 will revise this ASAP

Reply:
I might be missing something though, because desmos seems fine... Just want clarification.

Reply:
Oh, right. In your previous example, the minimum would actually be 1 still, since 10 / 5 = 2

Reply:
Okay yeah nevermind, I incorrectly applied that `min` in my calculations.